### PR TITLE
[interop_testing_te9] fix: level control server handles its own ticks

### PR DESCRIPTION
**Pull request by rochaferraz**
#### Problem & Solution
The all-clusters and lighting apps rely on the `af-gen-event.h` file for the `EMBER_AF_GENERATED_EVENTS` definition, along with its context. However, the file is not automatically generated by zap. Thus other sample apps that rely on level control don't perform the level transition.

This definition enables the cluster events/ticks that cause a transition from one state to another on time-dependent clusters such as level control.

This change to level control deprecates the use of af-events. Level control server cluster now handles its own timer for state transitions using standard Matter SDK calls.

This fixes the issue on `interop_testing_te9` event. Once discussed and approved another similar PR will be made against `master` branch.

#### Testing
- Tested with a 2-endpoint dimmable light example in chef app and chip-tool
- level control server cluster correctly transitions from 0->254 within the correct time span informed of 5s. Both on endpoints 1 and 2, that have level control clusters
- level control server cluster correctly transitions from 254->0 within the correct time span informed of 10s
- same tests as above, but interrupted during transition. So a transition that should perform a transition of 0->254 stops at around 120 by another command to perform a transition to 0, and correctly dims back to 0 on the informed transition time

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/app/clusters/level-control/level-control.cpp [hunk 1]
```diff
@@ -129,14 +131,22 @@ static void reallyUpdateCoupledColorTemp(EndpointId endpoint);
...
+static void timerCallback(System::Layer *, void * callbackContext)
+{
+    emberAfLevelControlClusterServerTickCallback(static_cast<EndpointId>(reinterpret_cast<uintptr_t>(callbackContext)));
+}
+
 static void schedule(EndpointId endpoint, uint32_t delayMs)
 {
-    emberAfScheduleServerTickExtended(endpoint, LevelControl::Id, delayMs, EMBER_AF_LONG_POLL, EMBER_AF_OK_TO_SLEEP);
+    DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(delayMs), timerCallback,
+                                          reinterpret_cast<void *>(endpoint));
```
Comment (bzbarsky-apple) [inline src/app/clusters/level-control/level-control.cpp:144]: Should probably cast through `uintptr_t` here too, for symmetry with the cast on the other side.

File: src/app/clusters/level-control/level-control.cpp [hunk 2]
```diff
@@ -129,14 +131,22 @@ static void reallyUpdateCoupledColorTemp(EndpointId endpoint);
...
 }
 
 static void deactivate(EndpointId endpoint)
 {
-    emberAfDeactivateServerTick(endpoint, LevelControl::Id);
+    DeviceLayer::SystemLayer().CancelTimer(timerCallback, reinterpret_cast<void *>(endpoint));
```
Comment (bzbarsky-apple) [inline src/app/clusters/level-control/level-control.cpp:149]: And here.

File: src/app/clusters/level-control/level-control.cpp [hunk 3]
```diff
@@ -510,6 +520,7 @@ bool emberAfLevelControlClusterStopWithOnOffCallback(app::CommandHandler * comma
 static EmberAfStatus moveToLevelHandler(EndpointId endpoint, CommandId commandId, uint8_t level, uint16_t transitionTimeDs,
                                         uint8_t optionMask, uint8_t optionOverride, uint16_t storedLevel)
 {
+
```
Comment (bzbarsky-apple) [inline src/app/clusters/level-control/level-control.cpp:523]:
```suggestion
```

**Review by bzbarsky-apple — COMMENTED** [review]
Like I said in #19392, as a followup we should remove all the emberAfLevelControlClusterServerTickCallback bits from all the af-gen-event.h files.

Also, why are both this and #19392 open?

**Pull request merged**